quic/s2n-quic-core/src/datagram/default.rs

```diff
@@ -0,0 +1,139 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// s2n-quic's default implementation of the datagram component
+
+use crate::{
+    datagram::{ConnectionInfo, Endpoint, Packet, Receiver, Sender},
+    event::Timestamp,
+};
+use alloc::collections::VecDeque;
+use bytes::Bytes;
+use core::time::Duration;
+
+#[derive(Debug, Default)]
+pub struct Disabled;
```

Comment:
It might be worth moving the `Disabled` provider to its own module, just so things don't get too big :)

Also, for these public structs, it's good practice to add a private field in them so they can only be created by `Default` or a builder. This ensures we can always add new fields and not break backward-compatibility:

```suggestion
pub struct Disabled(());
```